internal/acctest/config/oauth/openidconnect/settings/oauth_open_id_connect_settings_test.go

```diff
@@ -0,0 +1,202 @@
+package oauthopenidconnectsettings_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	client "github.com/pingidentity/pingfederate-go-client/v1200/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/acctest"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/acctest/common/pointers"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/provider"
+)
+
+type openIdConnectSettingsResourceModel struct {
+	defaultPolicyRef *client.ResourceLink
+	sessionSettings  *client.OIDCSessionSettings
+}
+
+func createDependenciesForTestBecauseIHaveTo(t *testing.T) {
```

Comment:
Could this be done with HCL instead of direct API calls? The way this is right now could leave dangling resources if a step fails in the middle, which ideally we would avoid.

Reply:
Resolving based on our discussion to rework tests in a separate branch.